### PR TITLE
feat: Reset route if neither the Viewer of the Sidebar is open

### PR DESCRIPTION
**Pull request by artonge**
When the viewer or the sidebar is opened, we add the fileid to the route.
When both of them are closed, we do not remove the fileid from the route.
This means that, upon reload, the sidebar will be opened even though it was closed previously.

This PR ensure that the fileid is removed from the route when both the Sidebar and the Viewer are closed.

https://github.com/user-attachments/assets/f363228c-d365-49e3-954c-0bbfaae19763

### CODE REVIEW

File: apps/files/src/components/FilesListVirtual.vue [hunk 1]
```diff
@@ -251,12 +256,31 @@ export default defineComponent({
 			}
 		},
 
+		handleSideBarCloseEvent() {
+			if (OCA.Viewer.file === '') {
```
Comment (susnux) [inline apps/files/src/components/FilesListVirtual.vue:260]: It feels wrong having access to `OCA.Viewer` here, is is not enough the check if `openFile` is set?
That would make it independent of the Viewer in this place.

Meaning the viewer should reset `openfile` on close.
So on sidebar close there are only two states:
1. `openfile` is set: do not remove the fileid
2. `openfile` is not set: also remove the fileid

Comment (artonge) [reply apps/files/src/components/FilesListVirtual.vue:260]: Makes sense as we are also relying on `openfile` for the other way.

**Review by susnux — APPROVED** [review]
makes sense (ideally add a cypress tests), only thing I am not sure about if this really should be inside the virtual file list. But code makes sense and seems to work fine.

File: apps/files/src/components/FilesListVirtual.vue [hunk 2]
```diff
@@ -83,6 +83,7 @@ import VirtualList from './VirtualList.vue'
 import logger from '../logger.ts'
 import FilesListTableHeaderActions from './FilesListTableHeaderActions.vue'
 import FileListFilters from './FileListFilters.vue'
+import { subscribe, unsubscribe } from '@nextcloud/event-bus'
```
[the remark this answers is not in the document]
Comment (artonge) [reply apps/files/src/components/FilesListVirtual.vue]: :people_hugging:

**Review by skjnldsv — APPROVED** [review]

**Review by skjnldsv — CHANGES_REQUESTED** [review]
Wait for it

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Backports ?

**Pull request merged**